frontend/src/lib/components/JSSnippet.tsx

```diff
@@ -22,7 +22,7 @@ function snippetFunctions(arrayJs = '/static/array.js'): string {
     }
     const snippetMethods = methods.join(' ')
 
-    return `!function(t,e){var o,n,p,r;e.__SV||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.crossOrigin="anonymous",p.async=!0,p.src=s.api_host.replace(".i.posthog.com","-assets.i.posthog.com")+"${arrayJs}",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="${snippetMethods}".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);`
+    return `!function(t,e){var o,n,p,r;e.__SV||(window.posthog && window.posthog.__loaded)||(window.posthog=e,e._i=[],e.init=function(i,s,a){function g(t,e){var o=e.split(".");2==o.length&&(t=t[o[0]],e=o[1]),t[e]=function(){t.push([e].concat(Array.prototype.slice.call(arguments,0)))}}(p=t.createElement("script")).type="text/javascript",p.crossOrigin="anonymous",p.async=!0,p.src=s.api_host.replace(".i.posthog.com","-assets.i.posthog.com")+"${arrayJs}",(r=t.getElementsByTagName("script")[0]).parentNode.insertBefore(p,r);var u=e;for(void 0!==a?u=e[a]=[]:a="posthog",u.people=u.people||[],u.toString=function(t){var e="posthog";return"posthog"!==a&&(e+="."+a),t||(e+=" (stub)"),e},u.people.toString=function(){return u.toString(1)+".people (stub)"},o="${snippetMethods}".split(" "),n=0;n<o.length;n++)g(u,o[n]);e._i.push([i,s,a])},e.__SV=1)}(document,window.posthog||[]);`
```

Comment:
this may well be the "correct" long term fix
(would be good to have an integration test in the posthog-js repo to prove it!)

but fixing it (only) here requires folk to edit their snippet

if we _also_ or instead fix it in array.js, then they'd get the fix when we release it

although arguably, anyone that has a working install doesn't need this fix so 🤔

Reply:
yeah I tried fixing it on `array.js` but couldn't get to a solution yet

at least updating the snippet will fix it for all future installs as well

I'll add the test on posthog-js, good idea

but, if this helps, mind approving so we can get this in? then I can also let the customer that opened that ticket to update their snippet

Reply:
all good no worriess

the main thing it does is add another truthy check with window.posthog.__loaded

so now is somethig like:

`e.__SV||(window.posthog && window.posthog.__loaded)||(window.posthog=BIG_FUNCTION_THAT_INITS_POSTHOG_AND_RETURNS_IT)`

before it was just:

`e.__SV||(window.posthog=BIG_FUNCTION_THAT_INITS_POSTHOG_AND_RETURNS_IT)`

because we didnt check for window.posthog, it means that if you add the script multiple times, you ended up with multiple instances sharing a bunch of references, causing the overflow

though the race condition might still matter, though it's already an issue today with the code as is

Reply:
hey @pauldambra I sent the updated snippet to the customer, looks like it indeed fixed that issue
given that, I'd love to get this in, unless you have any more thoughs or objections here?